src/Servers/HttpSys/src/IHttpSysRequestTimingFeature.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Server.HttpSys;
+
+/// <summary>
+/// This exposes the Http.Sys HTTP_REQUEST_TIMING_INFO.
+/// </summary>
+public interface IHttpSysRequestTimingFeature
+{
+    /// <summary>
+    /// Gets all http.sys timing timestamps
+    /// </summary>
+    /// <remarks>
+    /// These timestamps were originally retrieved using <see cref="System.Diagnostics.Stopwatch.GetTimestamp"/> and the timestamp frequency can be obtained via <see cref="System.Diagnostics.Stopwatch.Frequency"/>.
+    /// The index of the timing can be cast to <see cref="HttpSysRequestTimingType"/> to know what the timing represents.
+    /// The value may be 0 if the timing is not available for the current request.
+    /// </remarks>
+    IEnumerable<long> Timestamps { get; }
```

Comment:
Thoughts on making this IReadOnlyCollection instead?

Reply:
Ya I was going to but then I forgot that you can return a ReadOnlySpan because it's a ref struct (stack allocated). I think the reason for this is there is a concern around the lifetime of the backing array. For example, if people cache the span and stored it to process later after the request is completed, they will run into issues.

Reply:
You can return a `ReadOnlySpan` from a property. `ReadOnlyMemory<T>.Span` does exactly that. You're just limited in how you can use that returned value. You can always call `ToArray()` yourself though if you need to access it off the stack.

Reply:
Hmm good call it. The compiler was complaining but maybe I was doing something wrong. I try again.